src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -352,6 +351,11 @@ protected override void VisitSwitchSection(BoundSwitchSection node, bool isLastS
                                     if (inputSlot > 0)
                                     {
                                         MarkDependentSlotsNotNull(inputSlot, inputType, ref this.StateWhenFalse);
+                                        if (t.Syntax is RecursivePatternSyntax { Type: null, Designation: null, PositionalPatternClause: null } syntax &&
```

Comment:
>RecursivePatternSyntax [](start = 56, length = 22)

It is not correct to check the syntax.  See [https://github.com/dotnet/roslyn/blob/master/docs/compilers/Design/Bound%20Node%20Design.md#bound-nodes-should-capture-all-semantic-information-embedded-in-the-syntax](https://github.com/dotnet/roslyn/blob/master/docs/compilers/Design/Bound%20Node%20Design.md#bound-nodes-should-capture-all-semantic-information-embedded-in-the-syntax)